### PR TITLE
Torch export script to export `config.json` file to deployment directory

### PR DESCRIPTION
**Pull request by dbogunowicz**
# Testing

## Test 1 (`label_to_class` mapping automatically inferred from `dataset` argument (`imagenette`)

```bash
python export.py 
--dataset
imagenette
--dataset-path
/home/damian/imagenette
--checkpoint-path
zoo:cv/classification/resnet_v1-50/pytorch/sparseml/imagenette/pruned-conservative
```

results in:
```bash
...
2022-08-16 07:04:32 sparseml.pytorch.utils.exporter INFO     Created deployment folder at /home/damian/sparseml/pytorch_vision/resnet50_imagenette__35/deployment
INFO:sparseml.pytorch.utils.exporter:Created deployment folder at /home/damian/sparseml/pytorch_vision/resnet50_imagenette__35/deployment
2022-08-16 07:04:32 sparseml.pytorch.utils.exporter INFO     Saved model.onnx in the deployment folder at /home/damian/sparseml/pytorch_vision/resnet50_imagenette__35/deployment/model.onnx
INFO:sparseml.pytorch.utils.exporter:Saved model.onnx in the deployment folder at /home/damian/sparseml/pytorch_vision/resnet50_imagenette__35/deployment/model.onnx
2022-08-16 07:04:32 sparseml.pytorch.utils.exporter INFO     Created config.json file at /home/damian/sparseml/pytorch_vision/resnet50_imagenette__35/deployment
INFO:sparseml.pytorch.utils.exporter:Created config.json file at /home/damian/sparseml/pytorch_vision/resnet50_imagenette__35/deployment
```
and

```bash
(sparseml_venv) damian@lambdaquad:~/sparseml$ ls /home/damian/sparseml/pytorch_vision/resnet50_imagenette__35/deployment
config.json  model.onnx
```
`config.json`: `{"labels_to_class_mapping": {"0": "tench", "1": "English springer", "2": "cassette player", "3": "chain saw", "4": "church", "5": "French horn", "6": "garbage truck", "7": "gas pump", "8": "golf ball", "9": "parachute"}}
`

## Test 2 (where we set `label_to_class` mapping explicitly)

`dummy.json`: `{"0": "some", "1": "dummy", 2: "class"}`

```bash
python export.py 
--dataset
imagenette
--dataset-path
/home/damian/imagenette
--checkpoint-path
zoo:cv/classification/resnet_v1-50/pytorch/sparseml/imagenette/pruned-conservative
--labels-to-class_mapping dummy.json
```

results in:
```bash
...
2022-08-16 07:34:21 sparseml.pytorch.utils.exporter INFO     Created deployment folder at /home/damian/sparseml/pytorch_vision/resnet50_imagenette__60/deployment
INFO:sparseml.pytorch.utils.exporter:Created deployment folder at /home/damian/sparseml/pytorch_vision/resnet50_imagenette__60/deployment
2022-08-16 07:34:21 sparseml.pytorch.utils.exporter INFO     Saved model.onnx in the deployment folder at /home/damian/sparseml/pytorch_vision/resnet50_imagenette__60/deployment/model.onnx
INFO:sparseml.pytorch.utils.exporter:Saved model.onnx in the deployment folder at /home/damian/sparseml/pytorch_vision/resnet50_imagenette__60/deployment/model.onnx
2022-08-16 07:34:21 sparseml.pytorch.utils.exporter INFO     Created config.json file at /home/damian/sparseml/pytorch_vision/resnet50_imagenette__60/deployment
INFO:sparseml.pytorch.utils.exporter:Created config.json file at /home/damian/sparseml/pytorch_vision/resnet50_imagenette__60/deployment
2022-08-16 07:34:31 sparseml.pytorch.utils.exporter INFO     Appended labels_to_class_mapping data to config.json at /home/damian/sparseml/pytorch_vision/resnet50_imagenette__60/deployment/config.json
INFO:sparseml.pytorch.utils.exporter:Appended labels_to_class_mapping data to config.json at /home/damian/sparseml/pytorch_vision/resnet50_imagenette__60/deployment/config.json
```
and

```bash
(sparseml_venv) damian@lambdaquad:~/sparseml$ ls /home/damian/sparseml/pytorch_vision/resnet50_imagenette__60/deployment/
config.json  model.onnx
```
`config.json`: `{"labels_to_class_mapping": {"0": "some", "1": "dummy", "2": "class"}}
`

### CODE REVIEW

File: src/sparseml/pytorch/utils/exporter.py [hunk 1]
```diff
@@ -242,11 +244,19 @@ def export_torchscript(
         else:
             script_model(path, self._module)
 
-    def create_deployment_folder(self):
+    def create_deployment_folder(
+        self, labels_to_class_mapping: Optional[Union[str, Dict[int, str]]] = None
+    ):
         """
-        Create a deployment folder inside the `save_dir` directory.
+        Create a deployment folder inside the `config_path` directory.
```
Comment (dbogunowicz) [inline src/sparseml/pytorch/utils/exporter.py:251]:
```suggestion
        Create a deployment folder inside the `self._output_dir` directory.
```

File: src/sparseml/pytorch/image_classification/export.py [hunk 2]
```diff
@@ -149,6 +154,14 @@
     help="A path to a previous checkpoint to load the state from "
     "and resume the state for exporting",
 )
+@click.option(
+    "--labels_to_class-mapping",
+    "--labels-to-class_mapping",
+    type=str,
+    default=None,
+    help="Optional path to file containing the dataset-specific "
```
Comment (KSGulin) [inline src/sparseml/pytorch/image_classification/export.py:162]: nit: help string needs to be updated to match longer description string at top of file

**Review by KSGulin — COMMENTED** [review]
Commented a nit, otherwise LGTM

File: src/sparseml/pytorch/utils/exporter.py [hunk 3]
```diff
@@ -551,6 +566,82 @@ def export_onnx(
...
+        with open(config_file_path, "r") as outfile:
+            config = json.load(outfile.read())
+    else:
+        config = {}
+
+    # check whether the label names are not already present in the config.
+    if key_name in config.keys():
+        _LOGGER.warning(
+            f"The file: {CONFIG_JSON_NAME} already contains key {key_name}. "
+            f"The old {key_name} data will be overwritten..."
```
Comment (bfineran) [inline src/sparseml/pytorch/utils/exporter.py:619]:
```suggestion
            f"File: {CONFIG_JSON_NAME} already contains key {key_name}. "
            f" {key_name} data will be overwritten"
```

File: src/sparseml/pytorch/utils/exporter.py [hunk 4]
```diff
@@ -551,6 +566,82 @@ def export_onnx(
...
+
+    if isinstance(labels_to_class_mapping, str):
+        # deserialize .json into a dictionary
+        _, file_extension = os.path.splitext(labels_to_class_mapping)
+        if file_extension != ".json":
+            raise ValueError(
+                f"The path to the serialized `labels_to_class_mapping`is expected "
+                f"to point to a json file. "
+                f"However, found file with the extension {file_extension}"
+            )
```
Comment (bfineran) [inline src/sparseml/pytorch/utils/exporter.py:630]: not needed - if file format is invalid let json.load fail

**Review by bfineran — COMMENTED** [review]
LGTM pending nits

**Review by KSGulin — APPROVED** [review]
LGTM

**Review by bfineran — APPROVED** [review]

**Pull request merged**